From 864ce60461737423b4487390dd232caecbba4985 Mon Sep 17 00:00:00 2001
From: Matheus Machado Guilhermino <matheus.machadoguilhermino@windriver.com>
Date: Thu, 21 Apr 2022 20:16:47 +0000
Subject: [PATCH] Replace deprecated hiera function

Replaced the deprecated 'HieraPuppet.lookup()' function with the
'puppet lookup' command.

Signed-off-by: Matheus Machado Guilhermino <matheus.machadoguilhermino@windriver.com>
---
 lib/puppet/provider/openstack/auth.rb | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/lib/puppet/provider/openstack/auth.rb b/lib/puppet/provider/openstack/auth.rb
index 4026aec..99a8733 100644
--- a/lib/puppet/provider/openstack/auth.rb
+++ b/lib/puppet/provider/openstack/auth.rb
@@ -1,13 +1,12 @@
 #require 'puppet/provider/openstack/credentials'
 require File.join(File.dirname(__FILE__), '..','..','..', 'puppet/provider/openstack/credentials')
-require 'hiera_puppet'
 
 module Puppet::Provider::Openstack::Auth
 
   RCFILENAME = "/etc/platform/openrc"
 
   def lookup_hiera(key)
-    HieraPuppet.lookup(key, :undef, self, nil, :priority)
+    %(sudo puppet lookup #{key})[4...-1]
   end
 
   def get_admin_password
-- 
2.30.2

